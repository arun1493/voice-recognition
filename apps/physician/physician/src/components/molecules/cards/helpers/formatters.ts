import i18n from "i18next";
import _get from "lodash/get";
import { EMPTY_OBJECT } from "@suki-web/suki-business";

import {
  Allergy,
  AllergyReaction,
  CardMedicalHistoryProps,
  CardSurgicalHistoryProps,
  ParagraphCard,
  Problem
} from "../../../../types";
import { CardType } from "../../../../constants";

const formatParagraphCardData = (data: CardMedicalHistoryProps | CardSurgicalHistoryProps | any, cardType: string) => {
  const paragraphData: Array<ParagraphCard> = [];

  switch (cardType) {
    case CardType.ALLERGIES: {
      const { allergies, nkda } = data || EMPTY_OBJECT;

      allergies &&
        allergies.forEach((allergyItem: Allergy) => {
          const { allergen_name, reactions } = allergyItem || EMPTY_OBJECT;
          let allergyReactionContent = "";
          reactions &&
            reactions.forEach((allergyItem: AllergyReaction) => {
              const { reaction_name } = allergyItem;
              if (reaction_name) allergyReactionContent = allergyReactionContent.concat(reaction_name, ", ");
            });

          allergyReactionContent = allergyReactionContent.slice(0, -2); // to remove comma at the end of string

          if (allergen_name)
            paragraphData.push({
              title: allergen_name,
              content: allergyReactionContent
            });
        });

      if (nkda)
        paragraphData.push({
          content: i18n.t("pages.note.cards.errors.allergiesNkda")
        });
      break;
    }
    case CardType.PAST_MEDICAL_HISTORY: {
      const { active_problems } = data || EMPTY_OBJECT;
      active_problems &&
        active_problems.forEach((activeProblem: Problem) => {
          const description = _get(activeProblem, "note");
          const error = _get(activeProblem, "error");
          const title = _get(activeProblem, "diagnosis.icd_description");
          if (!error) paragraphData.push({ title, content: description });
        });
      break;
    }

    case CardType.PAST_SURGICAL_HISTORY: {
      data &&
        data.forEach((surgicalHistoryItem: Problem) => {
          const { description, note, error } = surgicalHistoryItem;
          if (!error) paragraphData.push({ title: description, content: note });
        });
      break;
    }

    default:
      break;
  }
  return paragraphData;
};

export { formatParagraphCardData };
