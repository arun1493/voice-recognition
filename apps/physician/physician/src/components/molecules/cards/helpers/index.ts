export * from "./labels";
export * from "./formatters";
