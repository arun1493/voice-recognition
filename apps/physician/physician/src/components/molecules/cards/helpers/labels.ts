import i18n from "i18next";

import { CardType, VITAL_SIGNS } from "../../../../constants";

const getCardTitle = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.titles.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.titles.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.titles.vitals");
    default:
      return "";
  }
};

const getVitalSign = (sign: string) => {
  switch (sign) {
    case VITAL_SIGNS.BLOOD_PRESSURE:
      return i18n.t("pages.note.cards.vitalSigns.bloodPressure");
    case VITAL_SIGNS.BMI:
      return i18n.t("pages.note.cards.vitalSigns.bmi");
    case VITAL_SIGNS.HEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.height");
    case VITAL_SIGNS.PULSE:
      return i18n.t("pages.note.cards.vitalSigns.pulse");
    case VITAL_SIGNS.WEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.weight");
    case VITAL_SIGNS.HEART_RATE:
      return i18n.t("pages.note.cards.vitalSigns.heartRate");
    case VITAL_SIGNS.RESPIRATORY_RATE:
      return i18n.t("pages.note.cards.vitalSigns.respiratoryRate");
    case VITAL_SIGNS.TEMPERATURE:
      return i18n.t("pages.note.cards.vitalSigns.temperature");
    default:
      return null;
  }
};

const getErrorMessage = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.errors.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.errors.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.errors.vitals");
    default:
      return "";
  }
};

export { getCardTitle, getErrorMessage, getVitalSign };
