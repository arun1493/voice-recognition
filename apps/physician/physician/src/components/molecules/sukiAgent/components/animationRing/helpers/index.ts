export * from "./ringSize";
export * from "./animation";
