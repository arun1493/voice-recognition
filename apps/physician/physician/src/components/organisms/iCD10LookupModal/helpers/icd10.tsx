import _isEmpty from "lodash/isEmpty";

const formatText = (text: string, highlight: string) => {
  if (_isEmpty(text)) return;
  const parts = text.split(new RegExp(`(${highlight})`, "gi"));
  return parts.map((part) => (part.toLowerCase() === highlight.toLowerCase() ? <strong>{part}</strong> : part));
};

const formatCode = (code: string) => {
  if (code?.length > 3) {
    return code?.substring(0, 3) + "." + code?.substring(3);
  }
  return code;
};

export { formatCode, formatText };
