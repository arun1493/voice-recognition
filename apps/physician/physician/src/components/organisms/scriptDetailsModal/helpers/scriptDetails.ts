import { ScreenSize } from "@suki-web/suki-ui";

const { SM, XS, XXS } = ScreenSize;

const getWidth = (breakPoint: ScreenSize) => {
  switch (breakPoint) {
    case XXS:
      return "100vw";
    case XS:
      return 484;
    case SM:
      return 664;
    default:
      return 782;
  }
};

const getHeight = (breakPoint: ScreenSize) =>
  breakPoint === XS || breakPoint === XXS ? "100vh" : "calc( 100vh - 56px )";

const getMaxHeight = (breakPoint: ScreenSize) => (breakPoint === XS || breakPoint === XXS ? "100vh" : 0);

export { getHeight, getMaxHeight, getWidth };
