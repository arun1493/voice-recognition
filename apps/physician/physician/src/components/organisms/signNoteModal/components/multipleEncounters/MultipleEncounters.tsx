import {
  FlexLeftColumn,
  FlexRightColumn,
  getMultipleEncounterWidth,
  Header,
  Modal,
  ScreenSize,
  Text,
  useBreakpoint
} from "@suki-web/suki-ui";
import { FunctionComponent } from "react";
import { format, isToday } from "date-fns";
import { useTranslation } from "react-i18next";

import {
  StyledButton,
  StyledEncounter,
  StyledMultipleEncountersContainer,
  StyledText
} from "./multipleEncounters.styled";
import { DATE_FORMAT } from "../../../../../constants";

type MultipleEncountersProps = {
  closeModal: VoidFunction;
  saveUnfinishedNote: VoidFunction;
  sendToEMR: (encounter: MockDataItem) => void;
  sendToEMRChart: VoidFunction;
};

type MockDataItem = {
  date: Date;
  doctor: string;
  id: string;
  visitType: string;
};

const mockData: MockDataItem[] = [
  {
    date: new Date("Jul 01 2021 11:27:19"),
    doctor: "Dr. Mary Adams",
    id: "first",
    visitType: "Admission"
  },
  {
    date: new Date("Jul 12 2022 14:07:39"),
    doctor: "Dr. John Barry",
    id: "second",
    visitType: "[Visit type]"
  }
];

const MultipleEncounters: FunctionComponent<MultipleEncountersProps> = ({
  closeModal,
  saveUnfinishedNote,
  sendToEMR,
  sendToEMRChart
}) => {
  const { t } = useTranslation();

  const breakpoint = useBreakpoint();

  const renderEncounter = (encounter: MockDataItem) => {
    const { date, doctor, id, visitType } = encounter;

    return (
      <StyledEncounter key={id} onClick={() => sendToEMR(encounter)}>
        <FlexLeftColumn>
          <Text isBold={true} size="md">
            {visitType}
          </Text>
          <Text size="sm">{doctor}</Text>
        </FlexLeftColumn>
        <FlexRightColumn>
          {isToday(date) && (
            <Text isBold={true} size="sm">
              {t("date.today")}
            </Text>
          )}
          {!isToday(date) && (
            <>
              <Text isBold={true} size="sm">
                {format(date, DATE_FORMAT.DDMMYYYY)}
              </Text>
              <Text size="sm">{format(date, DATE_FORMAT.hhmmA)}</Text>
            </>
          )}
        </FlexRightColumn>
      </StyledEncounter>
    );
  };

  const headerSize = breakpoint === ScreenSize.XXS ? "h6" : "h5";

  return (
    <Modal
      hasCloseButton={true}
      height="auto"
      onClose={closeModal}
      position="mc"
      testId="multiple-encounters-modal"
      width={getMultipleEncounterWidth(breakpoint)}
    >
      <StyledMultipleEncountersContainer>
        <Header level={headerSize}>{t("modals.signNoteModal.multipleEncounters.title")}</Header>
        {mockData.map(renderEncounter)}
        <StyledText>
          <Text>{t("modals.signNoteModal.multipleEncounters.text")}</Text>
        </StyledText>
        <StyledButton onClick={sendToEMRChart}>{t("modals.signNoteModal.multipleEncounters.submitToEMR")}</StyledButton>
        <StyledButton onClick={saveUnfinishedNote}>
          {t("modals.signNoteModal.multipleEncounters.saveToUnfinishedNotes")}
        </StyledButton>
      </StyledMultipleEncountersContainer>
    </Modal>
  );
};

export default MultipleEncounters;

export { MultipleEncounters };
