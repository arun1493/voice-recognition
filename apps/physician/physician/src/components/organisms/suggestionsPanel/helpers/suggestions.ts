import i18n from "i18next";

import { SuggestionTypes } from "../../../../constants";

const getTextBySuggestion = (suggestion: string) => {
  switch (suggestion) {
    case SuggestionTypes.PL:
      return i18n.t("components.suggestionsPanel.pl");
    case SuggestionTypes.ED:
      return i18n.t("components.suggestionsPanel.ed");
    default:
      return "";
  }
};

export { getTextBySuggestion };
