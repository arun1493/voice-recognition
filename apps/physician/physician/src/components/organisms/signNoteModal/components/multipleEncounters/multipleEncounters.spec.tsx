import _noop from "lodash/noop";
import ReactDOM from "react-dom";
import { ReactPortal } from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";

import { configureI18n } from "../../../../../helpers";
import { MultipleEncounters } from "./MultipleEncounters";
import { componentSetupWithStore } from "../../../../../hocs";
import { shouldMatchSnapshot } from "../../../../../helpers/snapshot.spec";

const setup = componentSetupWithStore(() => (
  <MultipleEncounters closeModal={_noop} saveUnfinishedNote={_noop} sendToEMR={_noop} sendToEMRChart={_noop} />
));

describe("MultipleEncounters modal", () => {
  let baseElement: Element;
  let element: Element | null;

  beforeEach(() => {
    configureI18n();
    baseElement = setup().baseElement;
    element = baseElement.querySelector("[data-cy='multiple-encounters-modal']");
    ReactDOM.createPortal = jest.fn<ReactPortal, any>((element, node) => element);
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    expect(baseElement).toBeTruthy();
  });

  it("should contain correct title", () => {
    expect(element?.innerHTML).toContain("Select the visit for this note");
  });

  it("should contain correct buttons text", () => {
    expect(element?.innerHTML).toContain("Submit to EMR chart");
    expect(element?.innerHTML).toContain("Save to unfinished notes");
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(MultipleEncounters);
  });
});
