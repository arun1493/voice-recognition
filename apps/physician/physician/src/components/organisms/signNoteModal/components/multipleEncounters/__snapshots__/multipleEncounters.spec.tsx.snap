// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleEncounters modal should match snapshot 1`] = `
Array [
  <div
    className="modalMaskstyled__StyledBackdrop-sc-169llx0-0 fRODrN"
    onClick={[Function]}
  />,
  <div
    className="modalstyled__StyledModal-sc-1u144rj-1 cbAisc"
    data-cy="multiple-encounters-modal"
    height="auto"
    width="560px"
  >
    <div
      aria-label="Close"
      className="modalstyled__StyledCloseButton-sc-1u144rj-0 bSHZet modal-close"
      height={56}
    >
      <div
        className="navigationIconstyled__StyledNavigationIcon-lwvkbw-0 imuqWK"
        data-cy=""
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="iconstyled__StyledButton-ifk1iv-0 kgKJBV"
          onClick={[Function]}
        >
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.70857 6.29289C7.31805 5.90237 6.68488 5.90237 6.29436 6.29289C5.90383 6.68342 5.90383 7.31658 6.29436 7.70711L10.5865 11.9993L6.2944 16.2914C5.90388 16.6819 5.90388 17.3151 6.2944 17.7056C6.68493 18.0961 7.31809 18.0961 7.70862 17.7056L12.0007 13.4135L16.2928 17.7056C16.6834 18.0961 17.3165 18.0961 17.7071 17.7056C18.0976 17.3151 18.0976 16.6819 17.7071 16.2914L13.4149 11.9993L17.7071 7.70711C18.0976 7.31658 18.0976 6.68342 17.7071 6.29289C17.3166 5.90237 16.6834 5.90237 16.2929 6.29289L12.0007 10.5851L7.70857 6.29289Z"
              fill="#999999"
              xmlns="http://www.w3.org/2000/svg"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="multipleEncountersstyled__StyledMultipleEncountersContainer-sc-3r72xs-1 iHQGft"
    >
      <h5
        className="headerstyled__StyledH5-sc-1y4xvz5-4 iITAex"
      >
        Select the visit for this note
      </h5>
      <div
        className="multipleEncountersstyled__StyledEncounter-sc-3r72xs-3 bRWNuY"
        onClick={[Function]}
      >
        <div
          className="styles__FlexLeftColumn-vrndfe-0 elqfwi"
        >
          <p
            className="textstyled__StyledText-sc-1vedkxi-0 gQoCmW"
            color="inherit"
            cursor="default"
            onClick={[Function]}
            size="md"
          >
            Admission
          </p>
          <p
            className="textstyled__StyledText-sc-1vedkxi-0 leaaTa"
            color="inherit"
            cursor="default"
            onClick={[Function]}
            size="sm"
          >
            Dr. Mary Adams
          </p>
        </div>
        <div
          className="styles__FlexRightColumn-vrndfe-1 eXOVVD"
        >
          <p
            className="textstyled__StyledText-sc-1vedkxi-0 kWpgJQ"
            color="inherit"
            cursor="default"
            onClick={[Function]}
            size="sm"
          >
            01/07/2021
          </p>
          <p
            className="textstyled__StyledText-sc-1vedkxi-0 leaaTa"
            color="inherit"
            cursor="default"
            onClick={[Function]}
            size="sm"
          >
            11:27 AM
          </p>
        </div>
      </div>
      <div
        className="multipleEncountersstyled__StyledEncounter-sc-3r72xs-3 bRWNuY"
        onClick={[Function]}
      >
        <div
          className="styles__FlexLeftColumn-vrndfe-0 elqfwi"
        >
          <p
            className="textstyled__StyledText-sc-1vedkxi-0 gQoCmW"
            color="inherit"
            cursor="default"
            onClick={[Function]}
            size="md"
          >
            [Visit type]
          </p>
          <p
            className="textstyled__StyledText-sc-1vedkxi-0 leaaTa"
            color="inherit"
            cursor="default"
            onClick={[Function]}
            size="sm"
          >
            Dr. John Barry
          </p>
        </div>
        <div
          className="styles__FlexRightColumn-vrndfe-1 eXOVVD"
        >
          <p
            className="textstyled__StyledText-sc-1vedkxi-0 kWpgJQ"
            color="inherit"
            cursor="default"
            onClick={[Function]}
            size="sm"
          >
            12/07/2022
          </p>
          <p
            className="textstyled__StyledText-sc-1vedkxi-0 leaaTa"
            color="inherit"
            cursor="default"
            onClick={[Function]}
            size="sm"
          >
            02:07 PM
          </p>
        </div>
      </div>
      <div
        className="multipleEncountersstyled__StyledText-sc-3r72xs-2 cMvITA"
      >
        <p
          className="textstyled__StyledText-sc-1vedkxi-0 fDyqSY"
          color="inherit"
          cursor="default"
          onClick={[Function]}
          size="md"
        >
          or
        </p>
      </div>
      <button
        className="multipleEncountersstyled__StyledButton-sc-3r72xs-0 kzuxIU"
      >
        Submit to EMR chart
      </button>
      <button
        className="multipleEncountersstyled__StyledButton-sc-3r72xs-0 kzuxIU"
      >
        Save to unfinished notes
      </button>
    </div>
  </div>,
]
`;
